### PR TITLE
C_F_POINTER

### PR DESCRIPTION
**Pull request by klausler**
Argument checking will be added before merging, if the general approach is found to be sound.

### CODE REVIEW

File: module/__fortran_builtins.f90 [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+! See https://llvm.org/LICENSE.txt for license information.
+! SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+!
+!------------------------------------------------------------------------------!
+
+! These naming shenanigans prevent names from Fortran intrinsic modules
+! from being usable on INTRINSIC statements, and force the program
+! to USE the standard intrinsic modules in order to access the
+! standard names of the procedures.
+module __Fortran_builtins
```
Comment (tskeith) [inline module/__fortran_builtins.f90]: I'm not sure I understand the comment. Is there anything that prevents user code from using `__Fortran_builtins` and calling `__builtin_c_f_pointer`? Should there be?

Comment (klausler) [reply module/__fortran_builtins.f90]: No, there isn't, and I'm not sure that there needs to be one -- it should be obvious that the intrinsics in this module are implementation artifacts.

**Comment by sscalpone** [comment]
I think there needs to be a dependency to make sure the _ file is compiled before the others.  Sorry for the quick note with no supporting evidence.

**Comment by klausler** [comment]
> I think there needs to be a dependency to make sure the _ file is compiled before the others. Sorry for the quick note with no supporting evidence.

I have no idea how to bring this about in the CMakeLists.txt file; I've added a comment that the module of builtins has to appear first.

**Comment by klausler** [comment]
Please take another look now that error checking and tests have been added.  (But not the dependence in CMakeLists.txt that @sscalpone points out is needed.)

File: test/semantics/c_f_pointer.f90 [hunk 2]
```diff
@@ -0,0 +1,30 @@
+! Enforce 18.2.3.3
+
+include '../../module/__fortran_builtins.f90'
+include '../../module/iso_c_binding.f90'
```
Comment (tskeith) [inline test/semantics/c_f_pointer.f90]: These includes aren't necessary. The error tests are run with a `-I` option that points to the predefined modules.

File: test/semantics/canondo16.f90 [hunk 3]
```diff
@@ -15,13 +15,11 @@
 
 ! CHECK: A DO loop should terminate with an END DO or CONTINUE
 
-module iso_fortran_env
-  type :: team_type
-  end type
-end
+include '../../module/__fortran_builtins.f90'
+include '../../module/iso_c_binding.f90'
```
Comment (tskeith) [inline test/semantics/canondo16.f90]: Unfortunately, the same isn't true here. We should probably convert all of these to be run with `test_errors.sh`.

**Comment by tskeith** [comment]
Looks good to me.

File: test/semantics/c_f_pointer.f90 (hunk 2 above)
Comment (klausler) [reply test/semantics/c_f_pointer.f90]: Ok.

**Pull request merged**